src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ImportService.cs

```diff
@@ -1,163 +1,126 @@
 using GovUk.Education.ExploreEducationStatistics.Common.Model;
-using GovUk.Education.ExploreEducationStatistics.Common.Services;
 using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
 using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
 using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
 using GovUk.Education.ExploreEducationStatistics.Data.Processor.Model;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
-using Microsoft.Extensions.Logging;
-using Newtonsoft.Json;
-using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationErrorMessages;
-using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationUtils;
-using static GovUk.Education.ExploreEducationStatistics.Common.TableStorageTableNames;
+using static GovUk.Education.ExploreEducationStatistics.Common.Services.FileStorageUtils;
+using static GovUk.Education.ExploreEducationStatistics.Data.Processor.Model.ImporterQueues;
 
 namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
 {
     public class ImportService : IImportService
     {
-        private readonly ContentDbContext _context;
-        private readonly IMapper _mapper;
-        private readonly ILogger _logger;
+        private readonly ContentDbContext _contentDbContext;
+        private readonly IImportRepository _importRepository;
+        private readonly IReleaseFileRepository _releaseFileRepository;
         private readonly IStorageQueueService _queueService;
-        private readonly ITableStorageService _tableStorageService;
         private readonly IUserService _userService;
 
-        public ImportService(ContentDbContext contentDbContext,
-            IMapper mapper,
-            ILogger<ImportService> logger,
+        public ImportService(
+            ContentDbContext contentDbContext,
+            IImportRepository importRepository,
+            IReleaseFileRepository releaseFileRepository,
             IStorageQueueService queueService,
-            ITableStorageService tableStorageService,
             IUserService userService)
         {
-            _context = contentDbContext;
-            _mapper = mapper;
-            _logger = logger;
+            _contentDbContext = contentDbContext;
+            _importRepository = importRepository;
+            _releaseFileRepository = releaseFileRepository;
             _queueService = queueService;
-            _tableStorageService = tableStorageService;
             _userService = userService;
         }
 
-        public async Task Import(Guid releaseId,
-            Guid subjectId,
-            string dataFileName,
-            string metaFileName,
-            IFormFile dataFile,
-            bool isZip)
+        public async Task<ImportStatus> GetStatus(Guid fileId)
```

Comment:
While the repository is also as service, it only contains simple operations, typically CRUD operations like we see in this case, `add`, `get, and `delete`. This is useful to simplify the main service which is concerned with joining together other services, doing security checks and transforming results to carry out the business logic. The repositories typically have little or no dependencies except for the database.

Any service which depends on a low level persistence operation that doesn't need extra transformation or a security check can depend on just the repository class without needing to satisfy all the other dependences of the main service.

An example of where the repository class comes in really useful here is in `CancelSpecificFileImportAuthorizationHandler` which is only concerned with getting the status of the import.

If `CancelSpecificFileImportAuthorizationHandler` depended on `ImportService` it would have to satisfy the dependency on `UserService` that `ImportService` has. This isn't possible since `UserService` depends on `IAuthorizationService` which in turn depends on all of the AuthorationHandlers, so a cyclic dependency error would occur.

The _passing-down-the-argument_ cases `DeleteImport` and `GetStatus` exist for convenience to avoid adding an additional dependency on the `ImportRepository` when a class is already depending on `ImportService`.

Reply:
Agreed, wish we were using repositories more if anything!